harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -230,7 +238,7 @@ def cleanup_iterator() -> None:
                 ) as load_path:
                     self._load(pathlib.Path(load_path))
 
-            if self.context.distributed.size > 1:
+            if self.context.distributed.size > 1 and self.use_horovod:
                 hvd.broadcast_parameters(self.context._main_model.state_dict(), root_rank=0)
                 for optimizer in self.context.optimizers:
                     hvd.broadcast_optimizer_state(optimizer, root_rank=0)
```

Comment:
why does ddp not need this step?

I mean, I understand this is a horovod-specific API, but why doesn't DDP need something equivalent?

Reply:
torch `DistributedDataParallel` does this in the constructor. it broadcasts the state_dict() from the local module at rank 0 to all the other ranks in the group.